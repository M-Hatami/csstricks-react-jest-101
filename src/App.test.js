import React from "react";
import ReactDOM from "react-dom";
import App from "./App";

// My Import here:
import { shallow } from "enzyme";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// My Tests here:
it("renders correctly", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it("calls componentDidMount", () => {
  jest.spyOn(App.prototype, "componentDidMount");
  const wrapper = shallow(<App />);
  expect(App.prototype.componentDidMount.mock.calls.length).toBe(1);
});
