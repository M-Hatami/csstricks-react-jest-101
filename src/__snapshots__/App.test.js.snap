// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="container"
        >
          <h2>
            CSSTricks Tests
          </h2>
          <div
            className="row"
          >
            <p>
              Loading...
            </p>
          </div>
        </div>,
        <footer>
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            React version: 
            16.7.0
          </div>
        </footer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              CSSTricks Tests
            </h2>,
            <div
              className="row"
            >
              <p>
                Loading...
              </p>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CSSTricks Tests",
            },
            "ref": null,
            "rendered": "CSSTricks Tests",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Loading...
              </p>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading...",
              },
              "ref": null,
              "rendered": "Loading...",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            React version: 
            16.7.0
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "React version: ",
              "16.7.0",
            ],
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "React version: ",
            "16.7.0",
          ],
          "type": "div",
        },
        "type": "footer",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="container"
          >
            <h2>
              CSSTricks Tests
            </h2>
            <div
              className="row"
            >
              <p>
                Loading...
              </p>
            </div>
          </div>,
          <footer>
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              React version: 
              16.7.0
            </div>
          </footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                CSSTricks Tests
              </h2>,
              <div
                className="row"
              >
                <p>
                  Loading...
                </p>
              </div>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CSSTricks Tests",
              },
              "ref": null,
              "rendered": "CSSTricks Tests",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Loading...
                </p>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Loading...",
                },
                "ref": null,
                "rendered": "Loading...",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              React version: 
              16.7.0
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "React version: ",
                "16.7.0",
              ],
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "React version: ",
              "16.7.0",
            ],
            "type": "div",
          },
          "type": "footer",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
